packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -22,14 +23,20 @@ import { isRunningOnCiServer } from "../util/ci-detection";
 import {
   hasConsentedTelemetry,
   writeTelemetryConsent,
+  hasPromptedForExtension,
+  writePromptedForExtension,
```

Comment:
I'd like to see these identifiers renamed to answer the question "_what_ extension?" Maybe `hasPromptedForHHVSCode`?